solana-programs/anchor/audius-data/lib/utils.ts

```diff
@@ -168,3 +172,33 @@ export const getContentNode = async (
     derivedAddress,
   };
 };
+
+/**
+ * Returns object containing
+ * content node delegate wallet address and 
+ * authority KeyPair based on spId; when ci=true
+ * returns hardcoded wallet address and new KeyPair
+ */
+export const getContentNodeWalletAndAuthority = async ({ spId, ci = false }) => {
+  const cnSpId = parseInt(spId);
+  let contentNodeAuthority;
+  let delegateWallet;
+  if (ci) {
+    contentNodeAuthority = anchor.web3.Keypair.generate();
+    delegateWallet = DUMMY_CN_WALLET_ADDRESSES[cnSpId - 1]
+  } else {
+    const spConfigFilePath = `creator-node/compose/env/tmp/shellEnv${cnSpId}.sh`
+    try {
+      const spConfig = await fs.readFile(path.join(process.env.PROTOCOL_DIR, spConfigFilePath), "utf-8");
```

Comment:
why not just invoke the shell script and use the environment variables? instead of manually decoding the *sh file

Reply:
it seemed like a lighter lift to read 3 files than to open a child process to execute 3 scripts and then read env vars from each subshell, but ymmv. this works, but i can change the implementation if needed.

Reply:
fair point, it's just more about adding additional code (decoding *.sh) instead of using the existing system (env propagation is already part of our e2e flow). however if this works, I don't think we should change it